--- a/ggml/src/ggml-quants.c
+++ b/ggml/src/ggml-quants.c
@@ -5329,12 +5329,12 @@
 
 static bool validate_float(float f, size_t i) {
     if (isinf(f)) {
-        fprintf(stderr, "ggml_validate_row_data: found inf value at block %zu\n", i);
+        GGML_LOG_ERROR("ggml_validate_row_data: found inf value at block %zu\n", i);
         return false;
     }
 
     if (isnan(f)) {
-        fprintf(stderr, "ggml_validate_row_data: found nan value at block %zu\n", i);
+        GGML_LOG_ERROR("ggml_validate_row_data: found nan value at block %zu\n", i);
         return false;
     }
 
@@ -5351,12 +5351,12 @@
 
 static bool validate_fp16(ggml_fp16_t f, size_t i) {
     if (isinf_fp16(f)) {
-        fprintf(stderr, "ggml_validate_row_data: found inf value at block %zu\n", i);
+        GGML_LOG_ERROR("ggml_validate_row_data: found inf value at block %zu\n", i);
         return false;
     }
 
     if (isnan_fp16(f)) {
-        fprintf(stderr, "ggml_validate_row_data: found nan value at block %zu\n", i);
+        GGML_LOG_ERROR("ggml_validate_row_data: found nan value at block %zu\n", i);
         return false;
     }
 
@@ -5365,7 +5365,7 @@
 
 static bool validate_e_e8m0(uint8_t e, size_t i) {
     if (e == 0xff) {
-        fprintf(stderr, "ggml_validate_row_data: found invalid e value %d at block %zu\n", e, i);
+        GGML_LOG_ERROR("ggml_validate_row_data: found invalid e value %d at block %zu\n", e, i);
         return false;
     }
 
@@ -5408,12 +5408,12 @@
 
 bool ggml_validate_row_data(enum ggml_type type, const void * data, size_t nbytes) {
     if (type < 0 || type >= GGML_TYPE_COUNT) {
-        fprintf(stderr, "%s: invalid type %d\n", __func__, type);
+        GGML_LOG_ERROR("%s: invalid type %d\n", __func__, type);
         return false;
     }
 
     if (nbytes % ggml_type_size(type) != 0) {
-        fprintf(stderr, "%s: invalid size %zu for type %s (type size = %zu)\n", __func__, nbytes, ggml_type_name(type), ggml_type_size(type));
+        GGML_LOG_ERROR("%s: invalid size %zu for type %s (type size = %zu)\n", __func__, nbytes, ggml_type_name(type), ggml_type_size(type));
         return false;
     }
 
@@ -5430,11 +5430,11 @@
                     infs += (f[i] & 0x7fff) == 0x7f80;
                 }
                 if (nans) {
-                    fprintf(stderr, "%s: found %d NaNs in row of %zu BF16 values\n", __func__, nans, nb);
+                    GGML_LOG_ERROR("%s: found %d NaNs in row of %zu BF16 values\n", __func__, nans, nb);
                     return false;
                 }
                 if (infs) {
-                    fprintf(stderr, "%s: found %d infinities in row of %zu BF16 values\n", __func__, infs, nb);
+                    GGML_LOG_ERROR("%s: found %d infinities in row of %zu BF16 values\n", __func__, infs, nb);
                     return false;
                 }
             } break;
@@ -5658,7 +5658,7 @@
             break;
         default:
             {
-                fprintf(stderr, "%s: invalid type %d\n", __func__, type);
+                GGML_LOG_ERROR("%s: invalid type %d\n", __func__, type);
                 return false;
             }
     }
