--- a/src/ua_types.c
+++ b/src/ua_types.c
@@ -623,7 +623,8 @@
     /* Convert to DateTime. Add or subtract one extra second here to prevent
      * underflow/overflow. This is reverted once the fractional part has been
      * added. */
-    sinceunix -= (sinceunix > 0) ? 1 : -1;
+    UA_Boolean positive = (sinceunix > 0);
+    sinceunix -= positive ? 1 : -1;
     UA_DateTime dt = (UA_DateTime)
         (sinceunix + (UA_DATETIME_UNIX_EPOCH / UA_DATETIME_SEC)) * UA_DATETIME_SEC;
 
@@ -675,7 +676,7 @@
 
  finish:
     /* Remove the underflow/overflow protection (see above) */
-    if(sinceunix > 0) {
+    if(positive) {
         if(dt > UA_INT64_MAX - UA_DATETIME_SEC)
             return UA_STATUSCODE_BADDECODINGERROR;
         dt += UA_DATETIME_SEC;
